parlai/scripts/train_model.py

```diff
@@ -103,12 +103,14 @@ def run_eval(agent, opt, datatype, max_exs=-1, write_log=False, valid_world=None
     print('[ running eval: ' + datatype + ' ]')
     if 'stream' in opt['datatype']:
         datatype += ':stream'
-    opt['datatype'] = datatype
-    if opt.get('evaltask'):
-        opt['task'] = opt['evaltask']
 
     if valid_world is None:
-        valid_world = create_task(opt, agent)
+        opt = opt.copy()
+        opt['datatype'] = datatype
+        if opt.get('evaltask'):
+            opt['task'] = opt['evaltask']
+        valid_agent = create_agent_from_shared(agent.share())
```

Comment:
hmm can we make this optional perhaps in case for models that it doesnt work for?

Reply:
I'll put this behind a (default False) flag for now